suite-native/assets/src/assetsSelectors.ts

```diff
@@ -73,25 +82,66 @@ export const selectDeviceAssetsWithBalances = (state: AssetsRootState & DeviceRo
         };
     });
 
-    return deviceNetworksWithAssets.map((networkSymbol: NetworkSymbol) => {
+    let totalFiatBalance = 0;
+
+    const assets = deviceNetworksWithAssets.map((networkSymbol: NetworkSymbol) => {
         const networkAccounts = accountsWithFiatBalance.filter(
             account => account.symbol === networkSymbol,
         );
         const assetBalance = networkAccounts.reduce(
-            (sum, { cryptoValue }) => sum.plus(cryptoValue),
-            new BigNumber(0),
+            (sum, { cryptoValue }) => sum + Number(cryptoValue),
+            0,
         );
         const fiatBalance = networkAccounts.reduce(
             (sum, { fiatValue }) => (fiatValue ? Number(fiatValue) + (sum ?? 0) : sum),
             null as number | null,
         );
 
+        totalFiatBalance += fiatBalance ?? 0;
+
         const asset: AssetType = {
             symbol: networkSymbol,
-            assetBalance: assetBalance.toFixed(),
+            // For assets we should always only 8 decimals to save space
+            assetBalance: assetBalance.toFixed(8),
```

Comment:
What space to save? Ethereum has 18 decimals, solana 9...

Reply:
Now I see that it's already rounded in AssetItem to 8 decimals and it's used only on the dashboard. It might be nice to define some constant for it and use it on both places so the connection is clear, but it's a nitpick.